File: AlcoolOuGasolina/app/src/main/java/com/example/alcoolougasolina/MainActivity.java
package com.example.alcoolougasolina;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {

    private TextInputEditText editPrecoAlcool, editPrecoGasolina;
    private TextView textResultado;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editPrecoAlcool = findViewById(R.id.editPrecoAlcool);
        editPrecoGasolina = findViewById(R.id.editPrecoGasolina);
        textResultado = findViewById(R.id.textResultado);
    }

    public void calcularPreco(View view)
    {
        //Recuperar valores digitados
        String precoAlcool = editPrecoAlcool.getText().toString();
        String precoGasolina = editPrecoGasolina.getText().toString();

        //Validar os campos digitados
        Boolean camposValidados = this.validarCampos(precoAlcool, precoGasolina);

        if(camposValidados)
        {
            //Converter string para números
            Double valorAlcool = Double.parseDouble(precoAlcool);
            Double valorGasolina = Double.parseDouble(precoGasolina);

            /*Fazer cálculo de menor preço
            *Se (valorAlcool / valorGasolina >= 0.7 é melhor utilizar gasolina
            * senão é melhor utilizar álcool
            **/
            Double resultado = valorAlcool / valorGasolina;
            if(resultado >= 0.7) textResultado.setText("Melhor utilizar Gasolina");
            if(resultado < 0.7) textResultado.setText("Melhor utilizar Álcool");
        }
        if(!camposValidados) textResultado.setText("Preencha os campos primeiro!");
    }

    public Boolean validarCampos(String pAlcool, String pGasolina)
    {
        Boolean camposValidados = true;

        if(pAlcool == null || pAlcool.equals("")) camposValidados = false;
        if(pGasolina == null || pGasolina.equals("")) camposValidados = false;

        return camposValidados;
    }
}

File: AppAlcoolOuGasolinaComKotlin/app/src/main/java/com/example/appalcoolougasolinacomkotlin/MainActivity.kt
package br.com.kotlin.cursoandroid.gasolinaoualcoolapp

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun calcularPreco(view: View){

        val usuario = Usuario()
        usuario.nome = "Jamilton Damasceno"

        textResultado.text = usuario.nome

        //recuperar valores digitados
        //val precoAlcool = findViewById<EditText>(R.id.editPrecoAlcool)
        val precoAlcool = editPrecoAlcool.text.toString()
        val precoGasolina = editPrecoGasolina.text.toString()
        /*
        val validaCampos = validarCampos(precoAlcool, precoGasolina)
        if( validaCampos ){
            calcularMelhorPreco( precoAlcool, precoGasolina )
        }else {
            textResultado.text = "Preencha os preços primeiro!"
        }*/

    }

    fun calcularMelhorPreco(precoAlcool: String, precoGasolina: String) {

        //Converte valores string para numeros
        val valorAlcool = precoAlcool.toDouble()
        val valorGasolina = precoGasolina.toDouble()

        /*Faz cálculo ( precoAlcool / precoGasolina )
            * Se resultado >= 0.7 melhor utilizar gasolina
            * senão melhor utilizar Álcool
        * */
        val resultadoPreco = valorAlcool / valorGasolina

        if( resultadoPreco >= 0.7 ){
            textResultado.text = "Melhor utilizar Gasolina!"
        }else{
            textResultado.text = "Melhor utilizar Álcool!"
        }

    }

    fun validarCampos(precoAlcool: String, precoGasolina: String) : Boolean {

        var camposValidados: Boolean = true
        if( precoAlcool == null || precoAlcool.equals("") ){
            camposValidados = false
        }else if( precoGasolina == null || precoGasolina.equals("") ){
            camposValidados = false
        }

        return camposValidados

    }

}
